android/app/src/main/java/org/fossasia/openevent/viewmodels/FeedFragmentViewModel.java

```diff
@@ -0,0 +1,55 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.ViewModel;
+import android.content.res.Resources;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.facebook.FeedItem;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.SharedPreferencesUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+public class FeedFragmentViewModel extends ViewModel {
+
+    private List<FeedItem> feedLive;
+
+    public FeedFragmentViewModel() {
+        feedLive = new ArrayList<>();
+    }
+
+    public List<FeedItem> getFeedList() {
+        return feedLive;
+    }
+
+    public void clearFeedList() {
+        feedLive.clear();
+    }
+
+    public void addAllFeed(List<FeedItem> feed) {
+        feedLive.addAll(feed);
+    }
+
+    public void updateFBPageID(Resources resources) {
```

Comment:
No passing of Android specific classes, please. Calculate SharedPref in fragment and pass the result here